packages/ag-charts-community/src/chart/interaction/interactionManager.ts

```diff
@@ -4,17 +4,20 @@ import { Logger } from '../../util/logger';
 import { isFiniteNumber } from '../../util/type-guards';
 import { BaseManager } from './baseManager';
 
-type InteractionTypes =
-    | 'click'
-    | 'dblclick'
-    | 'contextmenu'
-    | 'hover'
-    | 'drag-start'
-    | 'drag'
-    | 'drag-end'
-    | 'leave'
-    | 'page-left'
-    | 'wheel';
+export const InteractionTypesArray = [
```

Comment:
Nit: As this isn't a type, maybe rename to `INTERACTION_TYPES` (`Array` seems redundant)?

Reply:
these are the places where I wonder if we're not abusing the language a bit just do to what enums provide for free. any objections about using an enum here?

Reply:
Generally I prefer strings than enums for several reasons:
1. Code readability: enums tend to be more verbose (`InteractionTypes.Leave` vs. `'leave'`)
2. Strings are more debuggable because the debugger just shows enums as numbers.
3. String unions typing is (apparently) more strictly enforced.

In fact, I just quickly tested out enums and it looks like the compiler does not distinguish between numbers and enums at all:

```typescript
enum MyTypeEnum {
    One = 1,
    Two = 2,
}

type RankedType<T extends MyTypeEnum> = {
    type: T;
    rank: number;
};

const myRank: RankedType<MyTypeEnum> = { type: 8, rank: MyTypeEnum.One };
```

This example replicates a realistic scenario where the programmer was accidentally swapped around the `type` and `rank` values. However, this code compiles just fine despite the fact that `8` is not a valid `MyTypeEnum`. This is definitely undesirable.